pilot/cmd/pilot-agent/options/security.go

```diff
@@ -107,6 +111,15 @@ func SetupSecurityOptions(proxyConfig *meshconfig.ProxyConfig, secOpt *security.
 		log.Infof("using credential fetcher of %s type in %s trust domain", credFetcherTypeEnv, o.TrustDomain)
 		o.CredFetcher = credFetcher
 	}
+
+	if UseGkeWorkloadCertificate(security.GkeWorkloadCertChainFilePath,
```

Comment:
This seems problematic unless its 100% guaranteed that the files will be present *before* the pod is started. Does the implementation guarantee that? In current usage, typically the files may be populated after the pod starts.

Reply:
GKE workload certificates are provided by a CSI driver.  The pod will not start until the CSI driver has successfully written the key, certificate, and trust anchor set to the pod filesystem.

Reply:
The PR has been revised such that if o.CAProviderName == security.GoogleGkeWorkloadCertificateProvider, the code will check the presence of GKE workload certificate files.